### ISSUE TITLE
Passing frame.dtypes to Frame constructors

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#300]
I'm not sure if this is a bug or a feature request, but it would be intuitive to allow the `dtypes` argument of `Frame`'s constructors to accept another frame's `dtypes` attribute. Currently doing so results in an error:

```python
a = sf.Frame.from_element(1, index=[1, 2], columns=['a', 'b'])
b = sf.Frame.from_records(a.iter_tuple(), dtypes=a.dtypes)
# ValueError: The truth value of a container is ambiguous. For a truthy indicator of non-empty status, use the `size` attribute.
```

**Comment by flexatone** [comment]
Thanks for this suggestion: I agree, this should work. I think it used to work before we 0.7, when `Series` were Truthy, but now we need to add some additional handling. I will add this. In the mean time, you can probably do `dtypes=dict(a.dtypes)`

**Comment by flexatone** [comment]
This fix is completed and will be available as part of the 0.7.13 release.

```python
>>> sf.Frame.from_records((('1', '2020-01'), ('2', '2019-05')), columns=('a', 'b'),  dtypes=sf.Series.from_items((('a', int), ('b', np.datetime64))) 
... )                                                                                                                                                                
<Frame>
<Index> a       b               <<U1>
<Index>
0       1       2020-01
1       2       2019-05
<int64> <int64> <datetime64[M]>
```
